### ISSUE TITLE
Gii model support for advanced relations

### ISSUE THREAD

**Issue by nineinchnick** [yiisoft/yii2#7830]
Given the following schema, gii fails to generate proper relations:

~~~sql
DROP TABLE IF EXISTS "schema1"."table1" CASCADE;
DROP TABLE IF EXISTS "schema1"."table2" CASCADE;
DROP TABLE IF EXISTS "schema2"."table1" CASCADE;
DROP TABLE IF EXISTS "schema2"."table2" CASCADE;
DROP SCHEMA IF EXISTS "schema1" CASCADE;
DROP SCHEMA IF EXISTS "schema2" CASCADE;

CREATE SCHEMA "schema1";
CREATE SCHEMA "schema2";

CREATE TABLE "schema1"."table1" (
  id serial primary key,
  a varchar(255) not null,
  fk1 integer not null,
  fk2 integer not null,
  fk3 integer not null,
  fk4 integer not null,
  UNIQUE (fk1, fk2),
  UNIQUE (fk3, fk4)
);

CREATE TABLE "schema1"."table2" (
  id serial primary key,
  b varchar(255) not null,
  fk1 integer not null,
  fk2 integer not null,
  UNIQUE (fk1, fk2)
);

CREATE TABLE "schema2"."table1" (
  id serial primary key,
  c varchar(255) not null,
  fk1 integer not null,
  fk2 integer not null,
  fk3 integer not null,
  fk4 integer not null,
  fk5 integer not null,
  fk6 integer not null,
  UNIQUE (fk5, fk6),
  CONSTRAINT t1_f12_fkey FOREIGN KEY (fk1, fk2) REFERENCES "schema1"."table1" (fk2, fk1),
  CONSTRAINT t1_f34_fkey FOREIGN KEY (fk3, fk4) REFERENCES "schema1"."table1" (fk4, fk3)
);

CREATE TABLE "schema2"."table2" (
  id serial primary key,
  d varchar(255) not null,
  fk1 integer not null,
  fk2 integer not null,
  fk5 integer not null,
  fk6 integer not null,
  UNIQUE (fk1, fk2),
  UNIQUE (fk5, fk6),
  CONSTRAINT t2_f12_fkey FOREIGN KEY (fk1, fk2) REFERENCES "schema1"."table1" (fk1, fk2),
  CONSTRAINT t2_f56_fkey FOREIGN KEY (fk5, fk6) REFERENCES "schema2"."table1" (fk5, fk6)
);

~~~

The issues are:

* gii does not detect relations when generating a single model
* gii does not generate relations to other schemas
* gii does not detect `hasOne` relations properly
* gii does not maintain order of fks

I'm going to prepare a PR for this.

**Comment by nineinchnick** [comment]
Invalid `hasOne` relations are described in #5098.

**Comment by nineinchnick** [comment]
Composite FK order for `pgsql` is reported and fixed at #7831.
